### PR TITLE
Chore/setup cicd

### PR DESCRIPTION
**Pull request by mosharaf13**
https://github.com/mosharaf13/google-scrapper-ruby/issues/3

## What happened 👀

- Set docker registry host on repo
- Set docker registry token on repo
- Set heroku api token on repo
- Change config DOCKER_REGISTRY_USERNAME and DOCKER_IMAGE to match with docker account

## Insight 📝

- Opening PR to check if heroku deployment workflow succeeds

### CODE REVIEW

**Comment by mosharaf13** [comment]
Me too

File: .github/workflows/deploy_heroku.yml
Comment (sanG-github) [inline .github/workflows/deploy_heroku.yml:1]: Will we deploy it to Heroku or another alternative provider Mosharaf? 😉 
(Currently Heruko is charged for the deployments, so we usually use `Fly.io` with the Hobby plan for the alternative)

**Review by sanG-github — COMMENTED** [review]
Could you also provide the deployed site? 🙌

**Review by sanG-github — COMMENTED** [review]
Please help to fill these fields, also provide more POC (Proof of changes), then we can assert the changes based on the provided information. 🙌 

image

File: .github/workflows/deploy_heroku.yml
Comment (mosharaf13) [reply .github/workflows/deploy_heroku.yml:1]: Ah okay. I have deployed to heroku just to see if everything works or not. If everything is ok, I am thingking about deploying to fly later.

Currently [staging](https://google-scrapper-ruby-staging.herokuapp.com/) is crashing. I am trying to find out why.

Also, you can see in conversation that danger generated 0.00% code coverage 😕 .

Comment (mosharaf13) [reply .github/workflows/deploy_heroku.yml:1]: Creating fly configuration now. I will provide you deployed site once it's done 😀

Comment (mosharaf13) [reply .github/workflows/deploy_heroku.yml:1]: Deployed project can be found [here](https://google-scrapper-ruby.fly.dev/)

Comment (mosharaf13) [reply .github/workflows/deploy_heroku.yml:1]: Hi @sanG, something weird happened. I changed the base branch from main to develop and the PR closed automatically and now there's no option to change the base or reopen the PR.

Comment (mosharaf13) [reply .github/workflows/deploy_heroku.yml:1]: > Hi @SAng, something weird happened. I changed the base branch from main to develop and the PR closed automatically and now there's no option to change the base or reopen the PR.

It turns out develop and chore/setup-cicd had unrelated history. But if that's the case, it shouldn't let me change the base branch or at least show me a warning 😕 .

Comment (mosharaf13) [reply .github/workflows/deploy_heroku.yml:1]: I have created  a new PR [here](https://github.com/mosharaf13/google-scrapper-ruby/pull/27). This time created the develop branch from main.

Comment (sanG-github) [reply .github/workflows/deploy_heroku.yml:1]: No problems, man 🙏

**Pull request closed without merging**